src/scripts/ui/GrandBazaarBuildingBody.tsx

```diff
@@ -61,16 +65,58 @@ function TradesTab({
 }): React.ReactNode {
    const [buyResourceFilter, setBuyResourceFilter] = useState<Resource | null>(savedBuyResourceFilter);
    const [sellResourceFilter, setSellResourceFilter] = useState<Resource | null>(savedSellResourceFilter);
+   const [nameResourceFilter, setNameResourceFilter] = useState<string>("");
+   const [nameBuyFilter, setNameBuyFilter] = useState<boolean>(true);
+   const [nameSellFilter, setNameSellFilter] = useState<boolean>(true);
 
    const availableResources = Array.from(availableResourcesSet).sort((a, b) =>
       Config.Resource[a].name().localeCompare(Config.Resource[b].name()),
    );
 
+   function toggleSearchNameFilter(type: "buy" | "sell") {
+      if (type === "buy") {
+         if (!nameSellFilter) setNameSellFilter(true);
+         setNameBuyFilter((prev) => !prev);
+      } else {
+         if (!nameBuyFilter) setNameBuyFilter(true);
+         setNameSellFilter((prev) => !prev);
+      }
+   }
+
    return (
       <>
          <article role="tabpanel" className="f1 column" style={{ padding: "8px", overflow: "auto" }}>
             <fieldset>
                <legend>{t(L.GrandBazaarFilters)}</legend>
+               <div className="row">
+                  <div style={{ width: "120px" }}>Search</div>
+                  <input
```

Comment:
It seems `<input>` element does not have correct styling applied. I recommend using `<button> element as it has the correct styling